### DISCUSSION TITLE
More "Sidenav" , "SwitchInput" & others options, New components to add.

### DISCUSSION THREAD

**Discussion by MahmoudTarek44** [primefaces/community#34] [PrimeNG] [answered]
1- I think compared to other UI libraries for #Angular or other front-end frameworks, #PrimeNg has fewer options or examples for the sidenav component.
I may suggest adding something ( such slim or slim+ mode in the templates ) to be available and more..
Also providing the ability for customization using templates just as other components would be great.

2- For the switch input, It would be nice to have the ability to customize inner styles, add icons, add text, or maybe control animation behaviour, or even change the switch itself for anything but a circle.

3- I think considering adding more features for the "Steps" component would be nice, It also has the minimum options.

4- For new components we can have:
  - Customizable "QR Code" component would be a nice addition.
  - Bottom navigation bar for mobile view also would be nice addition.

Maybe in other discussion I'll bring another ideas..

**Comment by cagataycivici** [comment, marked as answer]
Hi,

1) There are so many options for navigation so we cover it differently, more options are available in Blocks and Templates instead.

- PrimeNG Sidebar: https://primeng.org/sidebar
- PrimeBlocks: https://blocks.primeng.org/#/application/sidebarlayout
- Templates like Apollo: https://primefaces.org/apollo-ng

2) The upcoming pass through properties will resolve this for sure so you can customize the look and feel easily.

3) Yes, we sure can enhance it by reviewing other Stepper components out there.

4) Sure, once we get the new styling ready, we'll get back to feature development like new components, mobile enhancements, RTL and more.
